lib/src/stacked_table.rs

```diff
@@ -52,13 +52,12 @@ pub trait TableSegment {
     fn get_value<'a>(&'a self, key: &[u8]) -> Option<&'a [u8]> {
         if let Some(value) = self.segment_get_value(key) {
```

Comment:
I think you could do something like:

if let Some(value) = self.segment_get_value(key) {
  return Some(value);
}

let parent_file = self.segment_parent_file()?;
...

Reply:
Yes, that seems a bit better. Done.